### PR TITLE
How to Lookup Ip Addresses in Java

### LINKED ISSUE #3096: Collections in Java
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
How to Lookup IP Addresses Using Java Programming Language

### Introduction paragraph (2-3 paragraphs):
A node is a device that is connected to the internet. Nodes that are computers are referred to as hosts. Every host has a unique identification number that makes it possible for it to be identified when connected to the internet.

This unique number is referred to as an Internet address a.k.a IP address. Most IP addresses are four bytes long these are the IPV4 addresses. While recently, there are small but growing IP addresses that are 16 bytes long that is, IPV6 addresses. 

Since IP addresses are great for computers but are difficult for humans to remember, this led to the introduction of the Domain Name System (DNS). A DNS associates hostnames that humans can easily remember e.g., www. google.com with IP addresses (172.217.170.206).

With Java Programming Language it is possible to check IP addresses using InetAddress.

### Key takeaways: 
1. What is an Ip?
2. Concept of a DNS and how it works
3. How Java represents an IP address using InetAddress Class.
4. Exception Handling mechanisms that are implemented.

### References: 
N/A

### PR DESCRIPTION
**Pull request by apondi-911**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3096 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing. @apondi-911 
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. :rocket:
Thank you.

**Comment by hectorkambow** [comment]
@apondi-911 
Closing to clear up the queue while the require revisions are made

**Comment by apondi-911** [comment]
Can my article be opened?

**Comment by apondi-911** [comment]
Asking if my article can be opened

**Comment by Dawe-7** [comment]
@louisefindlay23
Could we run another plagiarism check before i begin the review.

cc: @hectorkambow

**Comment by apondi-911** [comment]
Not yet

On Sep 14, 2021 00:53, "Dawe Daniel" ***@***.***> wrote:

@louisefindlay23 <https://github.com/louisefindlay23>
Could we run another plagiarism check before i begin the review.

cc: @hectorkambow <https://github.com/hectorkambow>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/section-engineering-education/engineering-education/pull/3454#issuecomment-918606615>,
or unsubscribe

**Comment by ahmadmardeni1** [comment]
plagiarism check done ✅

**Comment by Dawe-7** [comment]
> Hi, @apondi-911. Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revise your article accordingly.
> 
> Your article is quite short and not as detailed as we usually accept (thus is similar to other tutorials of the same nature). Please add more detail in order to make this more original and unique.
> 
> [How to Look Up IP Address in Java - Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7059490/How.to.Look.Up.IP.Address.in.Java.-.Report.pdf)

@apondi-911 I noticed that you haven't made any changes to your article. The article is still short, consider adding more content then i can begin the review.

**Comment by apondi-911** [comment]
Noted

On Sep 14, 2021 17:39, "Dawe Daniel" ***@***.***> wrote:

Hi, @apondi-911 <https://github.com/apondi-911>. Upon running your article
through our 3rd party plagiarism checker it seemed to raise a few flags and
the % was higher than we typically accept. Please see attached PDF - and
revise your article accordingly.

Your article is quite short and not as detailed as we usually accept (thus
is similar to other tutorials of the same nature). Please add more detail
in order to make this more original and unique.

How to Look Up IP Address in Java - Report.pdf
<https://github.com/section-engineering-education/engineering-education/files/7059490/How.to.Look.Up.IP.Address.in.Java.-.Report.pdf>

@apondi-911 <https://github.com/apondi-911> I noticed that you haven't made
any changes to your article. The article is still short, consider adding
more content then i can begin the review.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/section-engineering-education/engineering-education/pull/3454#issuecomment-919215615>,
or unsubscribe

**Comment by Dawe-7** [comment]
@apondi-911 is this article ready for review ?. What's the update ?

**Comment by apondi-911** [comment]
Yes it's ready

File: content/articles/how-to-lookup-ip-addresses-in-java/index.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    InetAddress address = InetAddress.getByName("www.google.com");
+    System.out.println(address);
+}
+catch(UnknownHostException ex){
+    System.out.println("www.google.com not found");
+}
+
+```
+
+### Commonly used methods.
```
Comment (Dawe-7) [inline content/articles/how-to-lookup-ip-addresses-in-java/index.md:102]:
```suggestion
### Commonly used methods
```

File: content/articles/how-to-lookup-ip-addresses-in-java/index.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+Example:
+
+```java
+InetAddress address = InetAddress.getByName("172.217.170.164");
+
+```
+
+> Hostnames are more stable than IP addresses. Most services have lived at the same hostnames but have changed their IP addresses severally. When choosing between hostnames like `www.google.com` and IP addresses like `172.217.170.164`, always go with the hostname.
+
+#### public static InetAddress[]getAllByName(String hostName) throws UnknownHostException
```
Comment (Dawe-7) [inline content/articles/how-to-lookup-ip-addresses-in-java/index.md:138]: This should not be a subheading. If it is a note change it to the appropriate method to represent things the reader should note. You can use `>` to identify a note. Do the same for the rest.

**Review by Dawe-7 — CHANGES_REQUESTED** [review]
@apondi-911 nice article, i've left you a few suggestions to implement. Adding a prerequisites section for readers would be helpful.  Also, your hero image should be` hero.jpg` and not `hero image.jpg`.

**Comment by Dawe-7** [comment]
@apondi-911 ping me when all changes have been made.

**Comment by Dawe-7** [comment]
@apondi-911 i noticed you still don't have a prerequisites section for readers. Kindly add one.

**Comment by apondi-911** [comment]
🙋

**Review by Dawe-7 — APPROVED** [review]
@apondi-911, kindly go through the changes made to improve on your future articles.

@Kayere, this article is ready for final review.

**Comment by WanjaMIKE** [comment]
@apondi-911 Kindly, take note of the edits I have done across the article. Thanks

**Comment by WanjaMIKE** [comment]
Merged #3454 into master.

**Pull request merged**